gravitee-am-ui/src/app/domain/applications/application/advanced/factors/factors.component.ts

```diff
@@ -28,24 +29,32 @@ import { AuthService } from '../../../../../services/auth.service';
 })
 export class ApplicationFactorsComponent implements OnInit {
   private domainId: string;
+  private riskAssessment: any = undefined;
 
   application: any;
   formChanged = false;
   factors: any[];
   editMode: boolean;
-
   deviceIdentifiers: any[];
-
   mfa: any;
-
   mfaStepUpRule = '';
   rememberDevice: any = {};
-
   adaptiveMfaRule = '';
+  skipAdaptiveMfaRule = '';
   enrollment: any = {};
   selectedMFAOption = '';
-  private riskAssessment: any = {};
+  selectedChallengeMFAOption = '';
+  challengeActive: boolean;
+  skipChallengeRule: string;
```

Comment:
**thought**: Why did you add dedicated attribute here (ex: challengeActive & skipChallengeRule) instead of adding a Challenge structure as we have in the Java part ? It maybe more simple to follow if the FrontEnd and the Backend settings structures are the same IMO. wdyt ?

Reply:
Outdated. This is an output from the challenge component and the challenge structure is different than in the backend.